lib/Dialect/HW/HWOps.cpp

```diff
@@ -3198,6 +3198,142 @@ ParseResult HierPathOp::parse(OpAsmParser &parser, OperationState &result) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ExportableRefOp
+//===----------------------------------------------------------------------===//
+
+void ExportableRefOp::build(OpBuilder &odsBuilder, OperationState &odsState,
+                            StringAttr circuitName,
+                            FlatSymbolRefAttr moduleName, StringAttr refName,
+                            ArrayRef<Attribute> namePath) {
+  auto symName =
+      StringAttr::get(odsBuilder.getContext(),
+                      "exportableRef." + circuitName.getValue() + "." +
+                          moduleName.getValue() + "." + refName.getValue());
+  build(odsBuilder, odsState, symName, circuitName, moduleName, refName,
+        ArrayAttr::get(odsBuilder.getContext(), namePath));
+}
+
+LogicalResult ExportableRefOp::verifyInnerRefs(hw::InnerRefNamespace &ns) {
+  if (getNamepath().empty())
+    return emitOpError() << "the name path cannot be empty";
+
+  auto expectedModuleName = getModuleNameAttr().getAttr();
+  for (unsigned i = 0, s = getNamepath().size() - 1; i < s; ++i) {
+    hw::InnerRefAttr innerRef = getNamepath()[i].cast<hw::InnerRefAttr>();
+
+    if (expectedModuleName != innerRef.getModule())
+      return emitOpError() << "name path is incorrect. Expected module: "
+                           << expectedModuleName
+                           << " instead found: " << innerRef.getModule();
+    HWInstanceLike instOp = ns.lookupOp<HWInstanceLike>(innerRef);
+    if (!instOp)
+      return emitOpError() << "module: " << innerRef.getModule()
+                           << " does not contain any instance with symbol: "
+                           << innerRef.getName();
+    expectedModuleName = instOp.getReferencedModuleNameAttr();
+  }
+
+  // The instance path has been verified. Now verify the last element.
+  auto leafRef =
+      getNamepath()[getNamepath().size() - 1].cast<hw::InnerRefAttr>();
+  if (!ns.lookup(leafRef))
+    return emitOpError() << " operation with symbol: " << leafRef
+                         << " was not found ";
+
+  if (expectedModuleName != leafRef.getModule())
+    return emitOpError() << "name path is incorrect. Expected module: "
+                         << expectedModuleName
+                         << " instead found: " << leafRef.getModule();
+
+  return success();
+}
+
+void ExportableRefOp::print(OpAsmPrinter &p) {
```

Comment:
Are custom printer/parsers necessary or can we get by with the default ones if using a `HierPathOp`?